libs/common/src/platform/services/key-generation.service.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { mock } from "jest-mock-extended";
+
+import { KdfConfig } from "../../auth/models/domain/kdf-config";
+import { CsprngArray } from "../../types/csprng";
+import { CryptoFunctionService } from "../abstractions/crypto-function.service";
+import { KdfType } from "../enums";
+
+import { KeyGenerationService } from "./key-generation.service";
+
+describe("KeyGenerationService", () => {
+  let sut: KeyGenerationService;
+
+  const cryptoFunctionService = mock<CryptoFunctionService>();
+
+  beforeEach(() => {
+    sut = new KeyGenerationService(cryptoFunctionService);
+  });
+
+  describe("createKey", () => {
+    test.each([256, 512])(
+      "it should create a key of the specified length",
+      async (bitLength: 256 | 512) => {
+        cryptoFunctionService.aesGenerateKey
+          .calledWith(bitLength)
+          .mockResolvedValue(new Uint8Array(bitLength / 8) as CsprngArray);
+
+        const key = await sut.createKey(bitLength);
+
+        expect(key.key.length).toEqual(bitLength / 8);
```

Comment:
I think I wrote this test prior to moving here, but it hits me the wrong way atm.

```suggestion
        cryptoFunctionService.aesGenerateKey
          .calledWith(bitLength)
          .mockResolvedValue(new Uint8Array(bitLength / 8) as CsprngArray);

        const key = await sut.createKey(bitLength);
        expect(cryptoFunctionService.aesGenerateKey).toHaveBeenCalledWith(bitLength);
```

I think I'd modify name to like "it should delegate key creation to crypto service"

Reply:
Thought it kind of makes sense as it tested the public interface instead of underlying implementation, but this works 👍